plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb

```diff
@@ -127,6 +128,10 @@ def set_replica_target
       template[:spec][:replicas] = @doc.replica_target if template[:kind] == 'Deployment'
     end
 
+    def set_name
+      template[:metadata][:name] = @doc.kubernetes_role.resource_name
```

Comment:
This changes the name used in ```deployment.spec.template.metadata.name``` in the yaml given to kubernetes? So, basically, it's override what the user configures there?